pkg/agent/packetcapture/packetcapture_controller_test.go

```diff
@@ -489,30 +489,29 @@ func TestPacketCaptureControllerRun(t *testing.T) {
 	go pcc.Run(stopCh)
 	for _, item := range pcs {
 		t.Run(item.name, func(t *testing.T) {
-			assert.Eventually(t, func() bool {
+			assert.EventuallyWithT(t, func(c *assert.CollectT) {
 				result, err := pcc.crdClient.CrdV1alpha1().PacketCaptures().Get(context.Background(), item.pc.Name, metav1.GetOptions{})
-				if err != nil {
-					return false
-				}
+				require.NoError(c, err)
+				var startedStatus, completeStatus, uploadStatus metav1.ConditionStatus
 				for _, cond := range result.Status.Conditions {
-					if cond.Type == crdv1alpha1.PacketCaptureComplete && item.expectCompleteStatus != cond.Status {
-						return false
+					if cond.Type == crdv1alpha1.PacketCaptureStarted {
+						startedStatus = cond.Status
+					}
+					if cond.Type == crdv1alpha1.PacketCaptureComplete {
+						completeStatus = cond.Status
 					}
-					if cond.Type == crdv1alpha1.PacketCaptureFileUploaded && item.expectUploadStatus != cond.Status {
-						return false
+					if cond.Type == crdv1alpha1.PacketCaptureFileUploaded {
+						uploadStatus = cond.Status
 					}
 				}
+				assert.Equal(c, item.expectStartedStatus, startedStatus)
+				assert.Equal(c, item.expectUploadStatus, uploadStatus)
+				assert.Equal(c, item.expectCompleteStatus, completeStatus)
+				assert.Equal(c, item.expectUploadStatus, uploadStatus)
 				if item.expectCompleteStatus == metav1.ConditionTrue {
-					if result.Status.NumberCaptured != testCaptureNum {
-						return false
-					}
+					assert.Equal(c, testCaptureNum, result.Status.NumberCaptured)
 				}
-				// delete cr
-				err = pcc.crdClient.CrdV1alpha1().PacketCaptures().Delete(context.TODO(), item.pc.Name, metav1.DeleteOptions{})
-				return err == nil
```

Comment:
I guess this is just removed because it contributed nothing to the test?
do we have a test that checks that the local capture file is deleted when the CR is deleted?